collector/collector.go

```diff
@@ -58,6 +58,10 @@ var (
 
 func registerCollector(name string, builder collectorBuilder, perfCounterNames ...string) {
 	builders[name] = builder
+	addPerfCounterDependencies(name, perfCounterNames)
+}
+
+func addPerfCounterDependencies(name string, perfCounterNames []string) {
```

Comment:
I can see why this is necessary, but it is a bit of a layering violation. We should probably - in a separate PR - overhaul the startup order so commandline parsing is done before collector initialization so we don't need to muck around in the internals like this in collectors.

Reply:
Please see commit 2d015fa3b7aeb9a1f7d57b243065ba44a9fc8f72 containing renamed structs.

Reply:
Performing command-line parsing before collector initialization sounds good -- especially in the case of the mssql collector, as I/O will need to be performed, and deferring I/O and other operations that can fail to code that runs outside of an init context would be ideal. I tried to keep this change as minimal as possible to get this working.